### ISSUE TITLE
[java] RedundantFieldInitializer report for annotation field not explicitly marked as final

### ISSUE THREAD

**Issue by SUPERCILEX** [pmd/pmd#215]
**Rule Set:** `optimizations.html#RedundantFieldInitializer`

**Description:** PMD incorrectly reports a redundant field initializer for final fields in an interface.

**Code Sample demonstrating the issue:**

```java
@IntDef({MetricType.CHECKBOX, MetricType.COUNTER, MetricType.SPINNER, MetricType.NOTE})
@Retention(RetentionPolicy.SOURCE)
public @interface MetricType {
    int CHECKBOX = 0; // <<<<<<<< incorrectly reported as a redundant field initializer
    int COUNTER = 1;
    int SPINNER = 2;
    int NOTE = 3;
}
```

**Running PMD through:** Gradle

**Comment by jsotuyod** [comment]
@SUPERCILEX I confirm the bug report.

The rule is checking all field declarations not explicitly marked as final, even if they are part of an interface or annotation. I'll work on a fix tomorrow, should be easy enough.

**Comment by SUPERCILEX** [comment]
@jsotuyod Awesome!
